src/Action/Delivery/Create.php

```diff
@@ -26,7 +33,30 @@ public function __invoke(Delivery $data)
             throw new ValidationException($errors);
         }
 
-        $this->pricingManager->createOrder($data);
+        $useArbitraryPrice = $this->authorizationCheckerInterface->isGranted('ROLE_ADMIN') && $data->hasArbitraryPrice();
+
+        if ($useArbitraryPrice) {
+            $arbitraryPrice = new ArbitraryPrice(
+                $data->getArbitraryPrice()->getVariantName(),
+                $data->getArbitraryPrice()->getValue()
+            );
+            $this->pricingManager->createOrder(
+                $data,
+                ['pricingStrategy' => new UseArbitraryPrice($arbitraryPrice)]
+            );
+        } else {
+            $priceForOrder = new UsePricingRules();
+            try {
+                $this->pricingManager->createOrder($data, [
+                    'pricingStrategy' => $priceForOrder,
+                    'throwException' => false
+                ]);
+    
+            } catch (NoRuleMatchedException $e) {
```

Comment:
It's not needed, is it? `PricingManager` does not throw an exception when `'throwException' => false`

Reply:
not needed, it is because I switched between true and false

Reply:
cleaned